e2etest/e2etest_suite_test.go

```diff
@@ -70,3 +83,218 @@ func TestE2E(t *testing.T) {
 	gomega.RegisterFailHandler(framework.Fail)
 	ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "E2E Suite", r)
 }
+
+var _ = ginkgo.BeforeSuite(func() {
+	cs, err := framework.LoadClientset()
+	framework.ExpectNoError(err)
+	_, err = cs.CoreV1().ConfigMaps(testNameSpace).Get(context.TODO(), "config", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		err = configureMetalLB(cs)
+		framework.ExpectNoError(err)
+	} else if err == nil {
+		metallbCMExisted = true
+	} else {
+		framework.ExpectNoError(err)
+	}
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	if !metallbCMExisted {
+		cs, err := framework.LoadClientset()
+		framework.ExpectNoError(err)
+		err = deleteMetalLBConfiguration(cs)
+		framework.ExpectNoError(err)
+	}
+})
+
+// Run a BGP router in a container for all of the speakers to peer with.
+func configureFRR(cs clientset.Interface) error {
+	// We need the ip of each node in the cluster to place them in the BGP router configuration file
+	// Each node will peer with this router.
+	nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+
+	// TODO: The IP address handling will need updates to add support for IPv6.
```

Comment:
So this^ is required before merging?

Reply:
im not sure about IPv6 support status. this wasn't support in the dev-env configuration either.